src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderInput.tsx

```diff
@@ -172,6 +182,18 @@ export const LimitOrderInput = ({
     [handleFormSubmit],
   )
 
+  const assetFilterPredicate = useCallback((asset: Asset) => {
+    return (
+      asset.chainId !== KnownChainIds.SolanaMainnet &&
+      isEvmChainId(asset.chainId) &&
+      !isNativeEvmAsset(asset.assetId)
```

Comment:
preferably-blocking: left side can go away (`isEvmChainId && !isNativeEvmAsset` will already filter solana and other chains' assets)

Reply:
Can do 3fb23025c7c6cfaf7382ef1d9baff820ecafc723